sdk/src/connections/usb/linux/device_enumerator_linux.cpp

```diff
@@ -29,19 +29,53 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include "connections/usb/usb_utils.h"
 #include "device_enumerator_impl.h"
-#include "utils_linux.h"
+#include "usb_linux_utils.h"
+#include "utils.h"
 
 #include <dirent.h>
 #include <fcntl.h>
 #include <glog/logging.h>
 #include <linux/videodev2.h>
 #include <sys/stat.h>
 
-#define CLEAR(x) memset(&(x), 0, sizeof(x))
+using namespace std;
+
+static aditof::Status getAvailableSensors(int fd,
+                                          string &advertisedSensorData) {
+    int ret;
+    uint16_t bufferLength;
+
+    ret = UsbLinuxUtils::uvcExUnitReadBuffer(
+        fd, 4, 0, reinterpret_cast<uint8_t *>(&bufferLength),
+        sizeof(bufferLength));
+    if (ret < 0) {
+        LOG(WARNING)
+            << "Failed to read size of buffer holding sensors info. Error: "
+            << ret;
+        return aditof::Status::GENERIC_ERROR;
+    }
+
+    uint8_t *data = new uint8_t[bufferLength + 1];
```

Comment:
you could use `unique_ptr<uint8_t[]>`, no need to `delete[]` in two separate places afterwards. (pass .get() to the uvcExUnitReadBuffer)